src/Bundle/spec/Routing/OperationRouteFactorySpec.php

```diff
@@ -0,0 +1,200 @@
+                'resource' => 'app.dummy',
+            ],
+        ]);
+    }
+
+    function it_generates_index_routes(): void
+    {
+        $metadata = Metadata::fromAliasAndConfiguration('app.dummy', ['driver' => 'dummy_driver']);
+
+        $route = $this->create(
+            $metadata,
+            new Resource('app.dummy'),
+            new Index(),
+        );
+
+        $route->getPath()->shouldReturn('/dummies');
+        $route->getMethods()->shouldReturn(['GET']);
+        $route->getDefaults()->shouldReturn([
+            '_sylius' => [
+                'resource' => 'app.dummy',
+            ],
+        ]);
+    }
+
+    function it_generates_show_routes(): void
+    {
+        $metadata = Metadata::fromAliasAndConfiguration('app.dummy', ['driver' => 'dummy_driver']);
+
+        $route = $this->create(
+            $metadata,
+            new Resource('app.dummy'),
+            new Show(),
+        );
+
+        $route->getPath()->shouldReturn('/dummies/{id}');
+        $route->getMethods()->shouldReturn(['GET']);
+        $route->getDefaults()->shouldReturn([
+            '_sylius' => [
+                'resource' => 'app.dummy',
+            ],
+        ]);
+    }
+
+    function it_generates_update_routes(): void
+    {
+        $metadata = Metadata::fromAliasAndConfiguration('app.dummy', ['driver' => 'dummy_driver']);
+
+        $route = $this->create(
+            $metadata,
+            new Resource('app.dummy'),
+            new Update(),
+        );
+
+        $route->getPath()->shouldReturn('/dummies/{id}/edit');
+        $route->getMethods()->shouldReturn(['GET', 'PUT']);
+        $route->getDefaults()->shouldReturn([
+            '_sylius' => [
+                'resource' => 'app.dummy',
+            ],
+        ]);
+    }
+
+    function it_generates_delete_routes(): void
+    {
+        $metadata = Metadata::fromAliasAndConfiguration('app.dummy', ['driver' => 'dummy_driver']);
+
+        $route = $this->create(
+            $metadata,
+            new Resource('app.dummy'),
+            new Delete(),
+        );
+
+        $route->getPath()->shouldReturn('/dummies/{id}');
+        $route->getMethods()->shouldReturn(['DELETE']);
+        $route->getDefaults()->shouldReturn([
+            '_sylius' => [
+                'resource' => 'app.dummy',
+            ],
+        ]);
+    }
+
+    function it_generates_routes_with_templates(): void
+    {
+        $metadata = Metadata::fromAliasAndConfiguration('app.dummy', ['driver' => 'dummy_driver']);
+
+        $route = $this->create(
+            $metadata,
+            new Resource('app.dummy'),
+            new Show(template: 'dummy/show.html.twig'),
+        );
+
+        $route->getPath()->shouldReturn('/dummies/{id}');
+        $route->getMethods()->shouldReturn(['GET']);
+        $route->getDefaults()->shouldReturn([
+            '_sylius' => [
+                'resource' => 'app.dummy',
+                'template' => 'dummy/show.html.twig'
```

Comment:
I'm wondering if we really need this, it's a lot simpler for now due to existing system, but later we can directly have these information from the Operation itself and it will be a lot simpler route.